### PR TITLE
Improvement/DRP-21

### PR DESCRIPTION
**Pull request by ichvandi**
### PURPOSES :
Improve app configuration to be more modular

### CHANGES :
- Move dependencies.gradle to root, so it becomes visible in gradle scripts
- Add flavors.gradle to differentiate config every flavor
- Add common.gradle for storing common `android` section in every module gradle
- Update `app`, `data`, `domain` gradle to adjust with changes
- Change package to `com.deuvox.deuvoxapp`

### PR NOTES :
Please check the changes, are there any more improvements that can be made?

### TEST RESULT :
None 🙅

### CODE REVIEW

**Comment by ichsanachmad** [comment]
Please add your ticket url on purposes section

**Review by ichsanachmad — CHANGES_REQUESTED** [review]
Please add your ticket url on purposes section

File: gradle.properties [hunk 1]
```diff
@@ -18,4 +18,13 @@ android.useAndroidX=true
...
+kotlin.code.style=official
+
+
+# App Config
+FLAVORS_DIMENSION=deuvox
+
+
+# App URL
+BASE_URL="https://private-anon-f649764b47-deuvox.apiary-mock.com/"
+BASE_URL_DEV="https://private-anon-f649764b47-deuvox.apiary-mock.com/"
```
Comment (ichsanachmad) [inline gradle.properties:30]: never put url on this git, i'm gonna add some ignore properties

File: flavors.gradle [hunk 2]
```diff
@@ -0,0 +1,21 @@
+ext.flavors = {
+
+    flavorDimensions FLAVORS_DIMENSION
+
+    productFlavors {
```
Comment (ichsanachmad) [inline flavors.gradle:5]: is this flavor confirmed with BE team ? bcs i think if we need to add some flavors, we need to compare it with BE flavors too

File: common.gradle [hunk 3]
```diff
@@ -0,0 +1,25 @@
+apply plugin: 'kotlin-android'
```
Comment (ichsanachmad) [inline common.gradle:1]: rename this file to base.gradle, it will be good for the abstraction CMIIW

File: build.gradle [hunk 4]
```diff
@@ -1,5 +1,5 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
-apply from :'buildsystem/dependencies.gradle'
```
Comment (ichsanachmad) [inline build.gradle:2]: why you move it to root project ? i think we need to wrap it on the package for custom gradle

**Review by ichsanachmad — CHANGES_REQUESTED** [review]
Please review again, thanks

File: flavors.gradle (hunk 2 above)
Comment (ichsanachmad) [reply flavors.gradle:5]: development
preproduction
production

File: gradle.properties (hunk 1 above)
Comment (ichvandi) [reply gradle.properties]: noted 👍

File: flavors.gradle (hunk 2 above)
Comment (ichvandi) [reply flavors.gradle]: noted  👍

File: build.gradle (hunk 4 above)
Comment (ichvandi) [reply build.gradle:2]: moved to `buildSrc`

**Review by ichsanachmad — COMMENTED** [review]
check again

**Review by ichsanachmad — APPROVED** [review]
LGTM

**Pull request merged**